lambdas/backend-client/src/client.ts

```diff
@@ -0,0 +1,101 @@
+import axios from 'axios';
+import { ITemplateClient } from './types/template-client';
+import {
+  CreateTemplate,
+  Failure,
+  Success,
+  SuccessList,
+  UpdateTemplate,
+} from './types/generated';
+
+type ApiResponse = Failure | Success | SuccessList;
+
+const client = axios.create({
+  baseURL: process.env.TEMPLATE_API_URL,
+});
+
+const failure = (data: Failure) => ({
+  error: {
+    code: data.statusCode,
+    message: data.technicalMessage,
+    details: data.details,
+  },
+});
+
+export const templateClient: ITemplateClient = {
```

Comment:
Could this be a class? Put the axios instance as a class member, and pass the token into the constructor? 

I can see how this breaks the backend/frontend thing. And why you've got the token going into the backend methods and are doing the token parsing there. 

It'd be nice if you could instantiate the api client used in the ui with a token, and just pass the template data into `createTemplate` etc. And the backend implementation of `createTemplate` would have the same signature, and the implementation would be instantiated with `event.requestContext.authorizer`. 

This might be getting to be a big lot of changes I'm asking for though 😅

Reply:
Done. 

Good suggestion, thanks!